Calibration/HcalCalibAlgos/test/RecAnalyzerMinbias.cc

```diff
@@ -173,73 +196,104 @@ RecAnalyzerMinbias::RecAnalyzerMinbias(const edm::ParameterSet& iConfig) :
   
 RecAnalyzerMinbias::~RecAnalyzerMinbias() {}
 
-// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
 void RecAnalyzerMinbias::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
-  //The following says we do not know what parameters are allowed so do no validation
-  // Please change this to state exactly what you do use, even if it is no parameters
-  edm::ParameterSetDescription desc;
-  desc.setUnknown();
-  descriptions.addDefault(desc);
-}
-  
-void RecAnalyzerMinbias::beginJob() {
-
-  std::string hc[5] = {"Empty", "HB", "HE", "HO", "HF"};
-  char        name[700], title[700];
-  for(int idet=1; idet<=4; idet++){
-    sprintf(name, "%s", hc[idet].c_str());
-    sprintf (title, "Noise distribution for %s", hc[idet].c_str());
-    h_[idet-1] = fs_->make<TH1D>(name,title,48,-6., 6.);
-  }
-
-  for (unsigned int i=0; i<hcalID_.size(); i++) {
-    HcalDetId id = HcalDetId(hcalID_[i]);
-    int subdet   = id.subdetId();
-    sprintf (name, "%s%d_%d_%d", hc[subdet].c_str(), id.ieta(), id.iphi(), id.depth());
-    sprintf (title, "Energy Distribution for %s ieta %d iphi %d depth %d", hc[subdet].c_str(), id.ieta(), id.iphi(), id.depth());
-    double xmin = (subdet == 4) ? -10 : -1;
-    double xmax = (subdet == 4) ? 90 : 9;
-    TH1D*  hh   = fs_->make<TH1D>(name, title, 50, xmin, xmax);
-    histo_.push_back(hh);
-  };
 
-  if (!fillHist_) {
-    myTree_       = fs_->make<TTree>("RecJet","RecJet Tree");
-    myTree_->Branch("cells",    &cells,    "cells/I");
-    myTree_->Branch("mysubd",   &mysubd,   "mysubd/I");
-    myTree_->Branch("depth",    &depth,    "depth/I");
-    myTree_->Branch("ieta",     &ieta,     "ieta/I");
-    myTree_->Branch("iphi",     &iphi,     "iphi/I");
-    myTree_->Branch("mom0_MB",  &mom0_MB,  "mom0_MB/F");
-    myTree_->Branch("mom1_MB",  &mom1_MB,  "mom1_MB/F");
-    myTree_->Branch("mom2_MB",  &mom2_MB,  "mom2_MB/F");
-    myTree_->Branch("mom3_MB",  &mom3_MB,  "mom3_MB/F");
-    myTree_->Branch("mom4_MB",  &mom4_MB,  "mom4_MB/F");
-    myTree_->Branch("trigbit",  &trigbit,  "trigbit/I");
-    myTree_->Branch("rnnumber", &rnnumber, "rnnumber/D");
-  }
-  myTree1_      = fs_->make<TTree>("RecJet1","RecJet1 Tree");
-  myTree1_->Branch("rnnum_",   &rnnum_,    "rnnum_/D");
-  myTree1_->Branch("HBHEsize", &HBHEsize,  "HBHEsize/I");
-  myTree1_->Branch("HFsize",   &HFsize,    "HFsize/I");
-
-  myMap_.clear();
+  std::vector<int> iarray;
+  edm::ParameterSetDescription desc;
+  desc.add<bool>("RunNZS",    true);
+  desc.add<bool>("Noise",     false);
+  desc.add<double>("ELowHB",  4);
+  desc.add<double>("EHighHB", 100);
+  desc.add<double>("ELowHE",  4);
+  desc.add<double>("EHighHE", 150);
+  desc.add<double>("ELowHF",  10);
+  desc.add<double>("EHighHF", 150);
+  desc.addUntracked<double>("EMin",2.0);
+  desc.addUntracked<double>("RunMin",303441.5);
+  desc.addUntracked<double>("RunMax",304825.5);
```

Comment:
as above, a short comment on how these numbers were obtained would be good for future users of the code